src/stores/workspace/WorkspaceStore.ts

```diff
@@ -146,16 +146,27 @@ export default class WorkspaceStore {
 
 	@action
 	public onSavedItemSelect = (savedItem: EventTreeNode | EventAction | EventMessage) => {
-		this.graphDataStore.timestamp = isEventMessage(savedItem)
-			? getTimestampAsNumber(savedItem.timestamp)
-			: getTimestampAsNumber(savedItem.startTimestamp);
 		if (isEventMessage(savedItem)) {
 			this.viewStore.activePanel = this.messagesStore;
 			this.messagesStore.onSavedItemSelect(savedItem);
 		} else {
 			this.viewStore.activePanel = this.eventsStore;
 			this.eventsStore.onSavedItemSelect(savedItem);
 		}
+		const newWorkspace = this.workspacesStore.createWorkspace();
+		this.workspacesStore.addWorkspace(newWorkspace);
+		if (isEventMessage(savedItem)) {
+			newWorkspace.graphDataStore.timestamp = getTimestampAsNumber(savedItem.timestamp);
+			newWorkspace.messagesStore = this.messagesStore;
```

Comment:
you reassign messages and events store but `this.messagesStore.workspaceStore` and `this.messagesStore.graphStore` will still point to SearchWorkspace workspace and graphStore